clients/src/main/java/org/apache/kafka/common/metrics/Metrics.java

```diff
@@ -555,8 +555,19 @@ public synchronized void removeReporter(MetricsReporter reporter) {
 
     synchronized void registerMetric(KafkaMetric metric) {
         MetricName metricName = metric.metricName();
-        if (this.metrics.containsKey(metricName))
-            throw new IllegalArgumentException("A metric named '" + metricName + "' already exists, can't register another one.");
+
+        KafkaMetric existingMetric = this.metrics.get(metricName);
+        if (existingMetric != null) {
+            log.error("The metric " + metricName + " is being replaced since it had already been registered.");
```

Comment:
i'd add "please file this as a bug report"

Reply:
+1